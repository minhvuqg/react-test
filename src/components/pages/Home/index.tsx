import { FC } from 'react'

import { ProductList } from 'components/UI/organisms/ProductList'

import { IProps } from './type.d'
import { productList } from 'data'

import './style.scss'

const Home: FC<IProps> = (): JSX.Element => {
  return (
    <div className="home-page">
      <div className="-container">
        <p className="-text-medium">Recommend</p>
        <ProductList products={productList} />
      </div>
    </div>
  )
}

export default Home
