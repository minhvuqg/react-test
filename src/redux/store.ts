import { combineReducers } from 'redux'
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'

import productSlice from './reducers/productSlice'

const customizedMiddleware: any = getDefaultMiddleware({
  serializableCheck: false
})

const persistedReducer: any = combineReducers({
  cart: productSlice
})

export default configureStore({
  reducer: persistedReducer,
  middleware: customizedMiddleware
})
