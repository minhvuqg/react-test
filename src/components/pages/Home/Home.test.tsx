import TestRenderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import Home from '../Home'
import store from 'redux/store'

it('render home correctly', () => {
  const homeEl = TestRenderer.create(
    < Provider store={store} >
      <Home />
    </Provider >).toJSON()
  expect(homeEl).toMatchSnapshot()
})
