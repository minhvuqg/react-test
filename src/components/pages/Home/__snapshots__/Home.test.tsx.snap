// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render home correctly 1`] = `
<div
  className="home-page"
>
  <div
    className="-container"
  >
    <p
      className="-text-medium"
    >
      Recommend
    </p>
    <div
      className="product-list -card"
    >
      <div
        className="product-card"
      >
        <div
          className="image"
        >
          <img
            alt="Iphone"
            data-testid="img"
            src="/imgs/products/iphone.png"
          />
        </div>
        <span
          className="m-t-32"
          data-testid="name"
        >
          Iphone
        </span>
        <span
          className="m-t-8 -text-bold"
          data-testid="price"
        >
          $1,000.00
        </span>
        <div
          className="actions m-t-32"
        >
          <button
            className="-custom-btn m-r-16 "
            data-testid="moreInfoBtn"
            onClick={[Function]}
          >
            More info
          </button>
          <button
            className="-custom-btn undefined -primary"
            data-testid="addToCartBtn"
            onClick={[Function]}
          >
            Add to cart
          </button>
        </div>
      </div>
      <div
        className="product-card"
      >
        <div
          className="image"
        >
          <img
            alt="Macbook pro 14 inch 2021 32gb"
            data-testid="img"
            src="/imgs/products/macbook.jpg"
          />
        </div>
        <span
          className="m-t-32"
          data-testid="name"
        >
          Macbook pro 14 inch 2021 32gb
        </span>
        <span
          className="m-t-8 -text-bold"
          data-testid="price"
        >
          $2,000.00
        </span>
        <div
          className="actions m-t-32"
        >
          <button
            className="-custom-btn m-r-16 "
            data-testid="moreInfoBtn"
            onClick={[Function]}
          >
            More info
          </button>
          <button
            className="-custom-btn undefined -primary"
            data-testid="addToCartBtn"
            onClick={[Function]}
          >
            Add to cart
          </button>
        </div>
      </div>
      <div
        className="product-card"
      >
        <div
          className="image"
        >
          <img
            alt="Watch"
            data-testid="img"
            src="/imgs/products/watch.jpg"
          />
        </div>
        <span
          className="m-t-32"
          data-testid="name"
        >
          Watch
        </span>
        <span
          className="m-t-8 -text-bold"
          data-testid="price"
        >
          $500.00
        </span>
        <div
          className="actions m-t-32"
        >
          <button
            className="-custom-btn m-r-16 "
            data-testid="moreInfoBtn"
            onClick={[Function]}
          >
            More info
          </button>
          <button
            className="-custom-btn undefined -primary"
            data-testid="addToCartBtn"
            onClick={[Function]}
          >
            Add to cart
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
